### PR TITLE
8539 add remove repeat item

### PR DESCRIPTION
**Pull request by scttnlsn**

### CODE REVIEW

File: app/decorators/results/response_form_context.rb [hunk 1]
```diff
@@ -16,5 +16,31 @@ def add(*items)
...
       "response[root]" + path.map { |item| "[#{item}]" }.join
     end
+
+    def id
+      "response-root-" + path.join("-")
+    end
+
+    def indices
+      path.reject { |item| item == :children }
+    end
```
Comment (smoyth) [inline app/decorators/results/response_form_context.rb]: Won't need this method anymore.

File: app/decorators/results/response_form_context.rb [hunk 2]
```diff
@@ -16,5 +16,31 @@ def add(*items)
     def input_name
       "response[root]" + path.map { |item| "[#{item}]" }.join
     end
+
+    def id
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/decorators/results/response_form_context.rb]: Also change class name to FormPath or something and just make it store a list of indices (integers or `__INDEX__`).

File: app/assets/javascripts/views/response_form_repeat_view.js.coffee [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class ELMO.Views.ResponseFormRepeatView extends ELMO.Views.ApplicationView
```
Comment (smoyth) [inline app/assets/javascripts/views/response_form_repeat_view.js.coffee]: Check input naming of repeats.

File: app/decorators/results/response_form_context.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (scttnlsn) [reply app/decorators/results/response_form_context.rb]: Leaving the name as is since it now also incorporates `read_only` (as per another comment)

**Comment by smoyth** [comment]
Looks terrific. Thanks for changes.

I did a small refactor of the context class using double splat. Please take a look:

https://github.com/thecartercenter/elmo/pull/471/commits/4698d9b891749ecfe32d57a5d1edb9c9e2b5cc2d

Thanks!

**Pull request merged**